### PR TITLE
Make some ngql methods public

### PR DESCRIPTION
**Pull request by kalman**

### CODE REVIEW

File: go/ngql/query.go [hunk 1]
```diff
@@ -37,29 +37,33 @@ const (
...
+	rootType := NomsTypeToGraphQLType(rootNomsType, false, tm)
 
 	return graphql.NewObject(graphql.ObjectConfig{
 		Name: rootQueryKey,
 		Fields: graphql.Fields{
 			rootKey: &graphql.Field{
 				Type: rootType,
 				Resolve: func(p graphql.ResolveParams) (interface{}, error) {
-					return maybeGetScalar(rootValue), nil
+					return MaybeGetScalar(rootValue), nil
```
Comment (kalman) [inline go/ngql/query.go]: Not super happy about exposing `MaybeGetScalar`, but I need to. If you can suggest a better name then great

**Comment by kalman** [comment]
CC @rafael-atticlabs

File: go/ngql/query.go (hunk 1 above)
Comment (arv) [inline go/ngql/query.go]: Maybe{Wrap,Box}Scalar ?

Comment (arv) [inline go/ngql/query.go]: Sorry... that is not what it does

Comment (arv) [inline go/ngql/query.go]: Can't think of a better name...

**Pull request merged**